anchor-service/src/anchor.rs

```diff
@@ -76,15 +100,123 @@ pub struct TimeEventBatch {
     /// The anchor proof
     pub proof: Proof,
     /// The Time Events
-    pub time_events: TimeEvents,
+    pub raw_time_events: RawTimeEvents,
 }
 
 impl std::fmt::Debug for TimeEventBatch {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         f.debug_struct("TimeEventBatch")
             .field("merkle_nodes", &self.merkle_nodes)
             .field("proof", &self.proof)
-            .field("time_events", &self.time_events)
+            .field("raw_time_events", &self.raw_time_events)
             .finish()
     }
 }
+
+impl TimeEventBatch {
+    /// Try to convert the TimeEventBatch into a list of TimeEventInsertables
+    pub fn try_to_insertables(&self) -> Result<Vec<TimeEventInsertable>> {
+        info!(
+            "store anchor batch: proof={}, events={}",
+            self.proof.to_cid()?,
+            self.raw_time_events.events.len()
+        );
+        let events = self
+            .raw_time_events
+            .events
+            .iter()
+            .map(|(anchor_request, time_event)| {
+                self.build_time_event_insertable(time_event, anchor_request)
+            })
+            .collect::<Result<Vec<_>>>()?;
+        Ok(events)
+    }
+
+    /// Build a TimeEventInsertable from a RawTimeEvent and AnchorRequest
+    fn build_time_event_insertable(
+        &self,
+        time_event: &RawTimeEvent,
+        anchor_request: &AnchorRequest,
+    ) -> Result<TimeEventInsertable> {
+        let time_event_cid = time_event.to_cid().context(format!(
```

Comment:
are any of these failures retry-able? I'm wondering if we should we just log a warning and ignore them or drop them from the next batch/attempt?

Reply:
okay, we can come back to this in the future